safe_video/ui/app.py

```diff
@@ -1,37 +1,111 @@
-            self.images.append(destination_path + file.name)
-            self.page.add(img)
+            path =self.cache_path + file.name
+            if not os.path.exists(self.cache_path):
+                os.makedirs(self.cache_path)
+            shutil.copy(file.path, path)
+            [name, format] = file.name.split(".", 1)
+            self.images[name] = Image(cache_path=self.cache_path, original_path=file.path, name=name, format=format)
+            self.current_image = name
+            
+            self.preview_bar_ref.current.controls.append(PreviewImage(name, path, self.switch_image))
+            self.preview_bar_ref.current.update()
+
+        self.current_image_key = name
+        self.image_ref.current.controls = [ft.Container(image_src=path, image_fit=ft.ImageFit.CONTAIN, expand=True, margin=10)]
+        self.image_ref.current.update()
+    
+    def switch_image(self, info: ft.ControlEvent):
+        self.current_image_key = info.control.key
+        img = self.images[self.current_image_key]
+        self.image_ref.current.controls = [ft.Container(image_src=img.get_path(), image_fit=ft.ImageFit.CONTAIN, expand=True, margin=10)]
+        self.image_ref.current.update()
+
+    def export_callback(self, file_results: ft.FilePickerResultEvent):
+        if file_results.path is None: return
+        img = self.images[self.current_image_key]
+        self.preview_bar_ref.current.controls = [c for c in self.preview_bar_ref.current.controls if c.key != img.name]
+        self.preview_bar_ref.current.update()
+        export_path = file_results.path
+        if '.' not in export_path:
+            export_path += '.' + img.format
+        shutil.move(img.get_path(), export_path)
+
+    def settings_callback(self):
+        print('TODO: Settings')
 
     def build_page(self, page: ft.Page):
         self.page = page
-        Mypick = ft.FilePicker(on_result=self.upload_callback)
-        page.overlay.append(Mypick)
+        page.padding=0
+        page.spacing=0
+        page.bgcolor = self.colors.light
+        file_picker_open = ft.FilePicker(on_result=self.upload_callback)
+        file_picker_export = ft.FilePicker(on_result=self.export_callback)
+        page.overlay.append(file_picker_open)
+        page.overlay.append(file_picker_export)
         page.add(
+            ft.Container(ft.Row([
+                ft.Container(content=ft.IconButton(ft.icons.BLUR_ON, focus_color=self.colors.dark), width=50),
+                ft.ElevatedButton("Open Image", on_click=lambda _: file_picker_open.pick_files(file_type=ft.FilePickerFileType.IMAGE, allow_multiple=True), icon=ft.icons.FOLDER_OPEN),
+                ft.ElevatedButton("Export file", on_click=lambda _: file_picker_export.save_file(file_name=self.images[self.current_image_key].name), icon=ft.icons.SAVE_ALT),
+                ft.ElevatedButton("Blur all", on_click=lambda _: self.blur_callback(), icon=ft.icons.PLAY_ARROW),
```

Comment:
This needs to be changed, that the right function is being called.
```suggestion
                ft.ElevatedButton("Blur all", on_click=lambda _: self.settings_callback(), icon=ft.icons.PLAY_ARROW),
```